### PR TITLE
Add Average Recall Metric

### PR DESCRIPTION
**Pull request by HawkinsZhao**
Hi @ybubnov , sorry for the late, I am quite busy these days on my graduation project. I have added the average recall metric and unit test code. Would you like to have a look at it?

Also, I really curious about the line 65 of `tests/test_metrics.py`, why don't you use a Dense layer with softmax activation for classification? The outputs are really different.

By the way, I could also add support for average precision metric. Although I don't know what kind of problems should it be used.

### CODE REVIEW

**Comment by ybubnov** [comment]
Hi @HawkinsZhao, thank you very much for this PR, highly appreciate this!

File: tests/test_metrics.py [hunk 1]
```diff
@@ -59,7 +62,7 @@ def create_model(self, outputs, loss, metrics_fns):
         model = keras.models.Sequential()
         model.add(keras.layers.Activation(keras.backend.sin))
         model.add(keras.layers.Activation(keras.backend.abs))
-        model.add(keras.layers.Lambda(lambda x: K.concatenate([x]*outputs)))
+        model.add(keras.layers.Dense(outputs, activation='softmax'))
```
Comment (ybubnov) [inline tests/test_metrics.py]: I've made the ```Lambda``` output intentionally, since the ```Dense``` layer according to the documentation looks like this: 
```py
output = activation(dot(input, kernel) + bias)
```

In this equation ```kernel``` is matrix initialized with ```glorot_uniform``` method by default. 

Assuming this, the layer output is always randomized. I've tried using unit kernel initializer, but I've failed to configure this layer to make it always return the same result. This turns model into the stochastic system, which I don't like to deal with.

The stochastic property of the model results in the inability to reproduce tests, since the system is always unpredictable. What I want: randomized input and static model. Such configuration gives ability to validate metrics on different data, but still able to troubleshoot failures.

Comment (ybubnov) [reply tests/test_metrics.py]: If you agree with my assumptions, it would be nice to see ```Lambda``` output here.

Comment (HawkinsZhao) [reply tests/test_metrics.py]: Thanks, @ybubnov. I agree with you about the point on making the model more static. But I am afraid the current `Lambda` output won't work on my implementation. I dumped label and predict in this way, at `test_metrics.py` about line 80:

```python
x, y = samples_fn(100)
model.fit(x, y, epochs=10)

numpy.save('p', model.predict(x))
numpy.save('y', y)
```

And I found the predict outputs looks like:
```python
array([[0.9997261 , 0.9997261 ],
       [0.78493714, 0.78493714],
       [0.5770823 , 0.5770823 ],
       [0.21627988, 0.21627988],
       .......
```

As you can see, probabilities for each label is the same. This could work for other cast strategies because they use `K.round`  for a single column. But mine require a general `argmax`. So, it would give me different answers when compared to others.

Let's see. I just came up with an "imaginary" solution. We could rescale the outputs like this:
```python
outputs = 2
scale = [v + 1 for v in range(outputs)] # [1, 2]
```

How about adding a `softmax` activation after the `Lambda` layer, this will solve my problem and make the test cases behave more like the real world classification problem.

```suggestion
model.add(keras.layers.Lambda(lambda x: K.concatenate([x]*outputs)))
scale = [v + 1 for v in range(outputs)]
model.add(keras.layers.Lambda(lambda x: (0.5 - x) * scale + 1))
model.add(keras.layers.Softmax())
```

Comment (ybubnov) [reply tests/test_metrics.py]: Agree with your approach.

**Comment by HawkinsZhao** [comment]
Added. Feel free to merge.
